src/test/java/bio/terra/service/snapshot/SnapshotTest.java

```diff
@@ -285,4 +294,98 @@ public void snapshotByFullViewHappyPathTest() throws Exception {
         assertEquals("new snapshot has been created", snapshot.getName(), requestModel.getName());
         assertEquals("all 5 relationships come through", snapshot.getRelationships().size(), 5);
     }
+
+    @Test
+    public void snapshotACLTest() throws Exception {
+        DatasetModel dataset = dataRepoFixtures.getDataset(steward(), datasetId);
+
+        String datasetName = dataset.getName();
+
+        logger.info("---- Dataset ACLs before snapshot create-----");
+        int datasetAclCount = fetchSourceDatasetAcls(datasetName);
+
+        //-------------------Create Snapshot----------------
+        SnapshotRequestModel requestModel =
+            jsonLoader.loadObject("ingest-test-snapshot-fullviews.json", SnapshotRequestModel.class);
+        // swap in the correct dataset name (with the id at the end)
+        requestModel.getContents().get(0).setDatasetName(datasetName);
+        SnapshotSummaryModel snapshotSummary =
+            dataRepoFixtures.createSnapshotWithRequest(steward(),
+                datasetName,
+                profileId,
+                requestModel);
+        createdSnapshotIds.add(snapshotSummary.getId());
+        SnapshotModel snapshot = dataRepoFixtures.getSnapshot(steward(), snapshotSummary.getId());
+        assertEquals("new snapshot has been created", snapshot.getName(), requestModel.getName());
+        assertEquals("There should be 5 snapshot relationships", snapshot.getRelationships().size(), 5);
+
+        // fetch ACLs
+        logger.info("---- Dataset ACLs after snapshot create-----");
+        int datasetPlusSnapshotCount = retryACLUpdate(datasetName, datasetAclCount, ACLCheck.GREATERTHAN);
+        assertThat("There should be more ACLs on the dataset after snapshot create",
+            datasetPlusSnapshotCount, greaterThan(datasetAclCount));
+
+        //-----------delete snapshot------------
+        dataRepoFixtures.deleteSnapshot(steward(), snapshotSummary.getId());
+        logger.info("---- Dataset ACLs after snapshot delete-----");
+        int datasetMinusSnapshotACLCount = retryACLUpdate(datasetName, datasetAclCount, ACLCheck.EQUALTO);
+        assertEquals("We should be back to the same number of ACLs on the dataset after snapshot delete",
+            datasetAclCount, datasetMinusSnapshotACLCount);
+    }
+
+    private int fetchSourceDatasetAcls(String datasetName) throws Exception {
```

Comment:
Do you think it makes sense to return a list here?  It would allow to test individual acls

Reply:
Yes, that makes sense with how I named the method. Although, we never use anything besides the size. We could go through and compare the actual acls, but I'm wondering if that's overkill for this test and might make it more flaky. Thoughts? Glad to add a more robust comparison between the acls!